core/admin.py

```diff
@@ -44,6 +49,83 @@ def get_readonly_fields(self, request, obj=None):
         return self.readonly_fields
 
+    def get_urls(self):
+        urls = super(EventAdmin, self).get_urls()
+        my_urls = [
+            url(r'manage_organizers/$',
+                self.admin_site.admin_view(self.view_manage_organizers),
+                name='core_event_manage_organizers'),
+            url(r'add_organizers/$',
+                self.admin_site.admin_view(self.view_add_organizers),
+                name='core_event_add_organizers'),
+        ]
+        return my_urls + urls
+
+    def _get_future_events_for_user(self, request):
+        """
+        Retrieves a list of future events, ordered by name.
+        It's based on get_queryset, so superuser see all events, while
+        is_staff users see events they're assigned to only.
+        """
+        return self.get_queryset(request) \
+            .filter(date__gte=datetime.now()
+                    .strftime("%Y-%m-%d")).order_by('name')
+
+    def _get_event_from_get(self, request, all_events):
+        """
+        Retrieves a particular event from request.GET['event_id'], or
+        returns the first one from all events available to the user.
+        """
+        if 'event_id' in request.GET:
+            try:
+                return all_events.get(id=request.GET['event_id'])
+            except Event.DoesNotExist:
+                pass
+        else:
+            return all_events.first()
+
+    def view_manage_organizers(self, request):
+        """
+        Custom admin view that allows user to remove organizers from an event
+        """
+        all_events = self._get_future_events_for_user(request)
+        event = self._get_event_from_get(request, all_events)
+
+        if 'remove' in request.GET and event in all_events:
+            user = User.objects.get(id=request.GET['remove'])
+            if user in event.team.all():
+                event.team.remove(user)
+                messages.success(request, 'Organizer {} has been removed'.format(user.get_full_name()))
+            return redirect('/admin/core/event/manage_organizers/?event_id={}'.format(event.id))
+
+        return render(request, 'admin/core/event/view_manage_organizers.html', {
+            'all_events': all_events,
+            'event': event,
+        })
+
+    def view_add_organizers(self, request):
+        """
+        Custom admin view that allows user to add new organizer to an event
+        """
+        all_events = self._get_future_events_for_user(request)
+        event = self._get_event_from_get(request, all_events)
+
+        if request.method == 'POST':
+            form = AddOrganizerForm(all_events, request.POST)
+            if form.is_valid():
+                user = form.save()
+                messages.success(request,
+                    '{} has been added to your event, yay!'.format(user.get_full_name()))
+                return redirect('/admin/core/event/manage_organizers/?event_id={}'.format(event.id))
```

Comment:
It may be less confusing to return to the "add organizers page". They usually ask me to add a bunch of people at the same time.
There is also a problem with the redirection: if you have more than one event, it will redirect you to the first event in alphabetical order.